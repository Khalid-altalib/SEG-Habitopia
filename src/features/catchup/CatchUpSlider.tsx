import { HStack } from "native-base";
import React from "react";
import CatchUpBox from "./CatchUpBox";

type Props = {};

const CatchUpSlider = (props: Props) => {
  return (
    <HStack mt={5} justifyContent={"space-between"}>
      <CatchUpBox />
    </HStack>
  );
};

export default CatchUpSlider;
