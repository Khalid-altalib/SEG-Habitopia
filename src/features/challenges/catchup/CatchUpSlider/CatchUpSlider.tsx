import { HStack } from "native-base";
import React from "react";
import CatchUpBox from "../CatchUpBox/CatchUpBox";

type Props = {};

const CatchUpSlider = (props: Props) => {
  return (
    <HStack mt={5} justifyContent={"space-between"} testID={"catchUpSlider"}>
      <CatchUpBox isUserLiked={false} />
    </HStack>
  );
};

export default CatchUpSlider;
