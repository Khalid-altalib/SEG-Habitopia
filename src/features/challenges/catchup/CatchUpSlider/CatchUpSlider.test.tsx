import { render } from "@testing-library/react-native";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import TestingWrapperNavigation from "@app/testingWrapperWithNavigation";
import CatchUpSlider from "./CatchUpSlider";

describe("CatchUpSlider", () => {
  const mockStore = configureStore([thunk])();

  it("renders the catch-up slider correctly", async () => {
    const component = render(
      <TestingWrapperNavigation store={mockStore}>
        <CatchUpSlider />
      </TestingWrapperNavigation>
    );

    expect(component).toMatchSnapshot();
  });

  it("finds the catch-up slider", async () => {
    const component = render(
      <TestingWrapperNavigation store={mockStore}>
        <CatchUpSlider />
      </TestingWrapperNavigation>
    );

    const box = component.getByTestId("catchUpSlider");

    expect(box).toBeDefined();
  });
});
