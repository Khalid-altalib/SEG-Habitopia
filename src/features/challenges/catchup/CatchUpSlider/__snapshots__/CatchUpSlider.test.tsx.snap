// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatchUpSlider renders the catch-up slider correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <RNSScreenStack
      style={
        {
          "flex": 1,
        }
      }
    >
      <RNSScreen
        collapsable={false}
        forwardedRef={[Function]}
        gestureResponseDistance={
          {
            "bottom": -1,
            "end": -1,
            "start": -1,
            "top": -1,
          }
        }
        nativeBackButtonDismissalEnabled={false}
        onAppear={[Function]}
        onDisappear={[Function]}
        onDismissed={[Function]}
        onHeaderBackButtonClicked={[Function]}
        onNativeDismissCancelled={[Function]}
        onTransitionProgress={[Function]}
        onWillDisappear={[Function]}
        replaceAnimation="push"
        stackPresentation="push"
        style={
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        swipeDirection="horizontal"
      >
        <RNSScreenStackHeaderConfig
          backButtonInCustomView={false}
          backgroundColor="rgb(255, 255, 255)"
          color="rgb(0, 122, 255)"
          direction="ltr"
          disableBackButtonMenu={false}
          hidden={false}
          hideBackButton={false}
          largeTitleHideShadow={false}
          title="TestScreen"
          titleColor="rgb(28, 28, 30)"
          topInsetEnabled={false}
          translucent={false}
        />
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          style={
            {
              "flex": 1,
              "flexDirection": "column-reverse",
            }
          }
        >
          <View
            style={
              [
                {
                  "flex": 1,
                },
                {
                  "backgroundColor": "rgb(242, 242, 242)",
                },
                undefined,
              ]
            }
          >
            <View
              dataSet={{}}
              style={
                {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginTop": 20,
                }
              }
              testID="catchUpSlider"
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={false}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "opacity": 1,
                  }
                }
                testID="catchUpBox"
              >
                <View
                  dataSet={{}}
                  style={
                    {
                      "height": 150,
                      "paddingBottom": 20,
                      "paddingLeft": 20,
                      "paddingRight": 20,
                      "paddingTop": 20,
                      "width": 150,
                    }
                  }
                >
                  <Image
                    accessibilityLabel="Alternate Text"
                    alt="Alternate Text"
                    dataSet={{}}
                    onError={[Function]}
                    source={
                      {
                        "uri": "https://picsum.photos/200",
                      }
                    }
                    style={
                      {
                        "borderRadius": 12,
                        "height": 128,
                        "maxWidth": "100%",
                        "position": "absolute",
                        "width": 128,
                      }
                    }
                  />
                  <View
                    accessibilityRole="button"
                    accessible={true}
                    collapsable={false}
                    colorScheme="red"
                    dataSet={{}}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      {
                        "alignItems": "center",
                        "borderRadius": 12,
                        "flexDirection": "row",
                        "justifyContent": "center",
                        "paddingBottom": 10,
                        "paddingLeft": 10,
                        "paddingRight": 10,
                        "paddingTop": 10,
                        "position": "absolute",
                      }
                    }
                    testID="heartButton"
                  >
                    <Text />
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RNSScreen>
    </RNSScreenStack>
  </View>
</RNCSafeAreaProvider>
`;
